### PR TITLE
feat: add endpoint for firehose history

### PR DESCRIPTION
**Pull request by rohilsurana**

### CODE REVIEW

File: internal/server/v1/firehose/handlers.go [hunk 1]
```diff
@@ -443,6 +460,45 @@ func getFirehoseResource(ctx context.Context, client entropyv1beta1.ResourceServ
...
+		st := status.Convert(err)
+		if st.Code() == codes.NotFound {
+			return nil, errors.ErrNotFound.
+				WithMsgf(firehoseNotFound).
+				WithCausef(st.Message())
+		}
+		return nil, err
+	}
+
+	prevSpec := []byte{'{', '}'}
```
Comment (spy16) [inline internal/server/v1/firehose/handlers.go:475]: You can write this as `[]byte("{}")` as well btw.

File: swagger.yml [hunk 2]
```diff
@@ -565,6 +565,35 @@ paths:
           description: internal error
           schema:
             $ref: "#/definitions/ErrorResponse"
+  /projects/{projectSlug}/firehoses/{firehoseUrn}/alertPolicy:
```
Comment (spy16) [inline swagger.yml:568]: Path is wrong.

Comment (rohilsurana) [reply swagger.yml:568]: Fixed.

File: internal/server/v1/firehose/handlers.go (hunk 1 above)
Comment (rohilsurana) [reply internal/server/v1/firehose/handlers.go:475]: Done!

File: internal/server/v1/firehose/firehose.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ func Routes(r *mux.Router, client entropyv1beta1.ResourceServiceClient, shieldCl
 	// read APIs
 	r.Handle("/projects/{projectSlug}/firehoses", handleListFirehoses(client, shieldClient)).Methods(http.MethodGet)
 	r.Handle("/projects/{projectSlug}/firehoses/{urn}", handleGetFirehose(client)).Methods(http.MethodGet)
+	r.Handle("/projects/{projectSlug}/firehoses/{urn}/history", handleGetFirehoseHistory(client)).Methods(http.MethodGet)
```
Comment (spy16) [inline internal/server/v1/firehose/firehose.go:36]: what if we changed this to `/revisions`?

Comment (rohilsurana) [reply internal/server/v1/firehose/firehose.go:36]: Currently the datlantis frontend calls it history hence wanted to keep the name same as that.
Plus there is a small logic b/w the entropy revisions API and this API hence wanted to make sure that is expressed in some manner.

**Review by spy16 — APPROVED** [review]

**Pull request merged**